saleor/graphql/core/types/taxes.py

```diff
@@ -373,7 +380,13 @@ def calculate_checkout_discounts(checkout_info):
                 checkout = checkout_info.checkout
                 discount_name = checkout.discount_name
                 return (
-                    [{"name": discount_name, "amount": checkout.discount}]
+                    [
+                        {
+                            "name": discount_name,
+                            "amount": checkout.discount,
+                            "type": TaxableObjectDiscountTypeEnum.SUBTOTAL,
```

Comment:
I would recommend to add a comment here, something like: 
`# Type is always subtotal, as voucher on shipping is always added to shipping_price`
This would be helpful in case someone would look at this code, and is not familiar with current discount propagation.